src/scenarios/image/ImagePreview/ImagePreview.tsx

```diff
@@ -0,0 +1,224 @@
+          }
+        }
+      } catch (error) {
+        setImages([]);
+        if (error instanceof Error) {
+          logger.error(
+            'calling [fetchImageForEntityId]',
+            'output:',
+            error.message,
+          );
+        }
+      }
+    } else {
+      setImages([]);
+      logger.error('Invalid entity ID.');
+    }
+  };
+
+  const setCurrentImageData = (
+    _event: React.SyntheticEvent<HTMLElement>,
+    data: DropdownProps,
+  ): void => {
+    const imageId = data.value as string;
+    setCurrentImageId(imageId);
+
+    const selectedImage = images.find((image) => image.id === imageId);
+    setWidth(selectedImage?.width ?? '');
+    setHeight(selectedImage?.height ?? '');
+    setPath(selectedImage?.path ?? '');
+  };
+
+  return (
+    <Segment basic>
+      <Header size="huge">Image Preview</Header>
+      <Header size="small">
+        Required Services:
+        <Label>catalog-service</Label>
+        <Label>image-service</Label>
+      </Header>
+
+      <Divider />
+
+      <Container fluid>
+        <p>This scenario execution is split in to two steps.</p>
+
+        <p>
+          In the first step, you can enter an Entity ID and retrieve the images
+          associated with it. The images will be populated under the Image
+          dropdown.
+        </p>
+
+        <p>
+          In the second stage, you can find the inputs which can be used to
+          transform the image selected from the dropdown (pre-populated from
+          first step).
+        </p>
+
+        <p>
+          As a result, an URL with the transformed image will be logged(the
+          focal point defined in Management System will be already applied)
+        </p>
+      </Container>
+
+      <Divider />
+
+      <Grid divided>
+        <Grid.Column width={8}>
+          <Segment basic>
+            <Form>
+              <Form.Input
+                control={VariableSearch}
+                icon="id card outline"
+                label="Entity ID"
+                value={entityId}
+                setStateValue={setEntityId}
+              />
+
+              <Form.Button
+                primary
+                onClick={async () => {
+                  fetchImageForEntityId();
+                }}
+              >
+                Fetch Images for Entity ID
+              </Form.Button>
+
+              <Divider />
+
+              <Form.Dropdown
+                fluid
+                selection
+                label="Image"
+                placeholder="Select Image"
+                options={images.map((image) => {
+                  return {
+                    text: `${image.path
+                      .split('/')
+                      .pop()} (${image.type.toLocaleLowerCase()})`,
+                    value: image.id,
+                  };
+                })}
+                value={currentImageId}
+                onChange={setCurrentImageData}
+              ></Form.Dropdown>
+
+              <Form.Input
```

Comment:
This should be a numeric input

Reply:
agreed on not needing the "Variable Search" component for the simple Width/Height fields. There is almost no reason to do it like that, and instead a simple number input suffices.

However, I will still leave the states, as we need to use them in the Image URL that we print in the log area. (it was not done so earlier, I just added the w, h query params in the URL now)